test_system_file_never_expired failing when building wheels

The ``test_system_file_never_expired`` test is causing failures when building our wheels:

https://travis-ci.org/MacPython/astropy-wheels/builds/603556155?utm_medium=notification&utm_source=github_status

The test failure is:

```
______________ TestDefaultAutoOpen.test_system_file_never_expired ______________
self = <astropy.utils.iers.tests.test_leap_second.TestDefaultAutoOpen object at 0x7f31c09c2908>
tmpdir = local('/tmp/pytest-of-root/pytest-0/test_system_file_never_expired0')
    @pytest.mark.skipif(not os.path.isfile(SYSTEM_FILE),
                        reason=f'system does not have {SYSTEM_FILE}')
    def test_system_file_never_expired(self, tmpdir):
        self.remove_auto_open_files('erfa')
        with iers.conf.set_temp('system_leap_second_file', SYSTEM_FILE):
            ls = iers.LeapSeconds.open()
            assert ls.expires > self.good_enough
            assert ls.meta['data_url'] in (iers.IERS_LEAP_SECOND_FILE,
                                           SYSTEM_FILE)
    
            # Also check with a "built-in" file that is expired
            fake_file = make_fake_file('28 June 2017', tmpdir)
            iers.LeapSeconds._auto_open_files[0] = fake_file
            ls2 = iers.LeapSeconds.open()
            assert ls2.expires > Time.now()
>           assert ls2.meta['data_url'] == SYSTEM_FILE
E           AssertionError: assert 'https://hpie...ap_Second.dat' == '/usr/share/z...-seconds.list'
E             - https://hpiers.obspm.fr/iers/bul/bulc/Leap_Second.dat
E             + /usr/share/zoneinfo/leap-seconds.list
/venv/lib/python3.7/site-packages/astropy/utils/iers/tests/test_leap_second.py:241: AssertionError
=============================== warnings summary ===============================
```

This seems to happen only for the Linux builds.

@mhvk - any idea what could be causing this?

--- a/astropy/utils/iers/tests/test_leap_second.py
+++ b/astropy/utils/iers/tests/test_leap_second.py
@@ -80,7 +80,7 @@ class TestReading:
         ls = iers.LeapSeconds.open(SYSTEM_FILE)
         expired = ls.expires < Time.now()
         if expired:
-            pytest.skip("CI system leap second file is expired.")
+            pytest.skip("System leap second file is expired.")
         assert not expired
 
 
@@ -250,7 +250,7 @@ class TestDefaultAutoOpen:
         # we should not depend on CI keeping it up to date, but if it is,
         # we should check that it is used if possible.
         if (iers.LeapSeconds.open(SYSTEM_FILE).expires <= self.good_enough):
-            pytest.skip("CI system leap second file is expired.")
+            pytest.skip("System leap second file is expired.")
 
         self.remove_auto_open_files('erfa')
         with iers.conf.set_temp('system_leap_second_file', SYSTEM_FILE):

--- a/astropy/utils/iers/tests/test_leap_second.py
+++ b/astropy/utils/iers/tests/test_leap_second.py
@@ -264,6 +264,7 @@ class TestDefaultAutoOpen:
             iers.LeapSeconds._auto_open_files[0] = fake_file
             ls2 = iers.LeapSeconds.open()
             assert ls2.expires > Time.now()
+            assert ls2.meta['data_url'] == SYSTEM_FILE
 
     @pytest.mark.remote_data
     def test_auto_open_urls_always_good_enough(self):

--- a/astropy/utils/iers/tests/test_leap_second.py
+++ b/astropy/utils/iers/tests/test_leap_second.py
@@ -266,7 +266,6 @@ class TestDefaultAutoOpen:
             iers.LeapSeconds._auto_open_files[0] = fake_file
             ls2 = iers.LeapSeconds.open()
             assert ls2.expires > Time.now()
-            assert ls2.meta['data_url'] == SYSTEM_FILE
 
     @pytest.mark.remote_data
     def test_auto_open_urls_always_good_enough(self):

--- a/astropy/utils/iers/tests/test_leap_second.py
+++ b/astropy/utils/iers/tests/test_leap_second.py
@@ -13,7 +13,6 @@ from astropy.utils.data import get_pkg_data_filename
 
 
 SYSTEM_FILE = '/usr/share/zoneinfo/leap-seconds.list'
-ON_CI = 'CI' in os.environ or 'TRAVIS' in os.environ
 
 # Test leap_seconds.list in test/data.
 LEAP_SECOND_LIST = get_pkg_data_filename('data/leap-seconds.list')
@@ -80,7 +79,7 @@ class TestReading:
     def test_open_system_file(self):
         ls = iers.LeapSeconds.open(SYSTEM_FILE)
         expired = ls.expires < Time.now()
-        if ON_CI and expired:
+        if expired:
             pytest.skip("CI system leap second file is expired.")
         assert not expired
 
@@ -250,8 +249,7 @@ class TestDefaultAutoOpen:
         # We skip the test if the system file is on a CI and is expired -
         # we should not depend on CI keeping it up to date, but if it is,
         # we should check that it is used if possible.
-        if (iers.LeapSeconds.open(SYSTEM_FILE).expires <= self.good_enough
-                and ON_CI):
+        if (iers.LeapSeconds.open(SYSTEM_FILE).expires <= self.good_enough):
             pytest.skip("CI system leap second file is expired.")
 
         self.remove_auto_open_files('erfa')

